### ISSUE TITLE
[Proposal] Refining LLM Trading

### ISSUE THREAD

**Issue by WaffleMoff** [uchicago-nlp-course/winter-2026-students#16]
[NLP_Final_Project_Paper.pdf](https://github.com/user-attachments/files/25169862/NLP_Final_Project_Paper.pdf)

**Comment by qrdai** [comment]
## 1. strengths

1. a novel idea with real-world impact. it would be useful to the community if a trading dataset that describes the latest market trends is presented.

## 2. weaknesses

1. more details are needed, especially about technical setups (e.g., which LLMs to use), and your plan for analyzing the performance results. for example, what ablation study would you carry out to understand how and why a model performs this way on your trading benchmark? and what temporal reasoning techniques are you going to use to see their impact on model's trading performance?

2. more comparisons with prior works are needed. i believe there are quite a few existing works in benchmarking LLMs' trading capabilities. how's your benchmark different from them (in data sources/metrics, etc.)? and what research question do you want to address by saying "refining" LLM trading, given that the main workload in your proposal is about building a trading dataset instead of algorithm-level refinements?

## 3. next steps

1. in week 8, should leave more time to analyzing LLMs' results on your trading benchmark. also remember to update at least one new figure / plot / table in each weekly blog.

**Comment by WaffleMoff** [comment]
[NLP Project Doc.pdf](https://github.com/user-attachments/files/25475081/NLP.Project.Doc.pdf)

**Comment by qrdai** [comment]
## 1. strengths

1. glad to see that you learned to scrape the web through this project. 


## 2. weaknesses

1. i still haven’t seen clear technical pathways, especially about the agentic framework and the finetuning plan. it’s usually easier to finetune only one LLM on the oil price data that you have collected, or just construct a multi-agent framework where different agents serve different roles and purposes (e.g., memory/reasoning/tool calling, etc) and do not involve finetuning at all. it would be much harder to combine multi-agent frameworks with finetuning -- agentic training is exceptionally tricky for both sft and rl. for sft, it’s hard to get high-quality long trajectory training data that covers the behavior of each agent. for rl, multi-agent training can be a nightmare with its sparse rewards over the long, multi-agent trajectories. given that we only have less than two weeks left, i strongly recommend considering training only one LLM or constructing a training-free multi-agent framework, instead of aiming for both.

2. it would be more helpful if you could present concrete experimental results next week, including but not limited to dataset statistics, training curves, and evaluation results.

**Comment by WaffleMoff** [comment]
[NLP Project Doc.pdf](https://github.com/user-attachments/files/25672019/NLP.Project.Doc.pdf)

**Comment by qrdai** [comment]
thank you for bringing up more technical details, experimental results, and the corresponding analysis! I have a better understanding of the whole picture now, but there are still quite a few technical details I don’t have a good intuition for, which i suggest including in your final report.

1. what does your training data and evaluation data look like? any examples or templates for your data? what do models need to learn during training (i.e., which part is loss computed on in a training example), and how do models predict the answer during evaluation (i.e., how is “mean token accuracy” defined and calculated)? finally, what does training entropy stand for and how is it computed, since entropy is not a common metric used for SFT (which I assume you are using, instead of RL)? I strongly suggest including data examples in your final report to help us get a better intuition of the task.


2. your data construction process and the dataset statistics. How did you construct the dataset (potentially with what LLMs as the data curator, or with what rule-based procedures), and how did you manage to obtain the ground truth answer for each question? how many questions in total, and any statistics available (e.g., answer distributions, average length, etc.)?


3. details about training and eval setups, including the size and mode (instruct/thinking) of your Qwen3 model, and other training/eval hyperparams. it would be useful to get a detailed log of your hyperparam tuning process so as to inspire other researchers:)
